clang/lib/3C/DeclRewriter.cpp

```diff
@@ -706,10 +716,17 @@ FunctionDeclBuilder::buildDeclVar(PVConstraint *IntCV, PVConstraint *ExtCV,
       buildItypeDecl(ExtCV, Decl, Type, IType, RewriteParm, RewriteRet);
     return;
   }
-  // Variables that do not need to be rewritten fall through to here. Type
-  // strings are taken unchanged from the original source.
-  if (isa<ParmVarDecl>(Decl)) {
-    Type = getSourceText(Decl->getSourceRange(), *Context);
+  // Variables that do not need to be rewritten fall through to here.
+  // For parameter variables, we try to extract the declaration from the source
+  // code. This preserves macros and other formatting. This isn't possible for
+  // return variables because the itype on returns is located after the
+  // parameter list. Sometimes the source range for a parameter declaration is
+  // not valid, for example if a function prototype is declared using a typedef.
+  // For these cases, we just fall back to reconstructing the declaration from
+  // the PVConstraint.
+  SourceRange Range = Decl->getSourceRange();
+  if (isa<ParmVarDecl>(Decl) && Range.isValid()) {
+    Type = getSourceText(Range, *Context);
```

Comment:
If we want to address the example I originally posted [here](https://github.com/correctcomputation/checkedc-clang/pull/434#pullrequestreview-593726037):
```c
#define MYMACRO int x, int y, int *
int foo(MYMACRO z) {
  return *z;
}
```
this would be the place.  When the parameter declaration is inside a macro, the range is "valid" but `getSourceText` still returns the empty string.  We would apply a similar fallback, except we have to preserve the parameter name in case it is part of a function definition that uses the parameter.  Here's [my implementation](https://github.com/correctcomputation/checkedc-clang/compare/fix_430...params-in-macro) on top of this PR.  If we think the value merits the extra code, you can incorporate the change into this PR or I can submit a later PR.